scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/SparkeyUriSideInput.scala

```diff
@@ -0,0 +1,66 @@
+package com.spotify.scio.extra.sparkey
+
+import java.util
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{EmptyTap, EmptyTapOf, ScioIO, Tap, TapOf}
+import com.spotify.scio.util.RemoteFileUtil
+import com.spotify.scio.values.SCollection
+import com.spotify.sparkey.{IndexHeader, LogHeader, SparkeyReader}
+import org.apache.beam.sdk.options.PipelineOptions
+import com.google.protobuf.ByteString
+
+
+object SparkeyUriSideInput {
+  object ReadP {
+    def apply(options: PipelineOptions): ReadP = ReadP(() => RemoteFileUtil.create(options))
+  }
+  case class ReadP(rfuCreateFn: () => RemoteFileUtil)
+
+  private trait TestSparkeyReader extends SparkeyReader {
+    override def getAsString(key: String): String = ???
+    override def getAsByteArray(key: Array[Byte]): Array[Byte] = ???
+    override def getAsEntry(key: Array[Byte]): SparkeyReader.Entry = ???
+    override def getIndexHeader: IndexHeader = ???
+    override def getLogHeader: LogHeader = ???
+    override def duplicate(): SparkeyReader = ???
+    override def close(): Unit = ???
+    override def iterator(): util.Iterator[SparkeyReader.Entry] = ???
+  }
+
+  def forTest[T](items: Map[String, String]): SparkeyUri = new SparkeyUri {
+    override val basePath: String = "i should be ignored"
+    override private[sparkey] def exists = true
+    override def getReader: SparkeyReader = new TestSparkeyReader {
+      override def getAsString(key: String): String = items.get(key).orNull
+    }
+  }
+
+  def forTest(items: Map[Array[Byte], Array[Byte]])(implicit d: DummyImplicit): SparkeyUri = new SparkeyUri {
```

Comment:
`Array[Byte]` shouldn't be used as map key. I doubt there'll be any collision but still bad practice.

Is the `DummyImplicit` to avoid method overloading issues? Maybe we have to use `Seq[(Array[Byte], Array[Byte])]` after all and that'll solve both issues but a bit inconsistent.

Alternatively maybe rename them to `fromStrings/fromBytes`?

Reply:
Ack, good point. Using `Seq` resolves both issues.